packages/paste-core/components/menu/__tests__/menu-badge-customization.spec.tsx

```diff
@@ -0,0 +1,136 @@
+      const menuButton = screen.getByTestId('menu-badge');
+      expect(menuButton.getAttribute(PASTE_ELEMENT)).toEqual('MENU_BADGE_BUTTON');
+      expect(menuButton.parentElement?.parentElement?.getAttribute(PASTE_ELEMENT)).toEqual('MENU_BADGE');
+    });
+
+    it('Should add the correct "data-paste-element" attribute when element prop is "CUSTOM_MENU"', () => {
+      render(<BaseMenu element="CUSTOM_MENU" />);
+
+      const menuButton = screen.getByTestId('menu-badge');
+      expect(menuButton.getAttribute(PASTE_ELEMENT)).toEqual('CUSTOM_MENU_BADGE_BUTTON');
+      expect(menuButton.parentElement?.parentElement?.getAttribute(PASTE_ELEMENT)).toEqual('CUSTOM_MENU_BADGE');
+    });
+  });
+
+  describe('Custom styles', () => {
+    describe('Style overrides using the default element name', () => {
+      it('Should apply correct style rules', () => {
+        render(<BaseMenu menuBadgeVariant="decorative10" />, {
+          wrapper: initWrapper('menu'),
+        });
+
+        const nodeCalledMenuBadge = screen.getByTestId('menu-badge');
+        expect(nodeCalledMenuBadge).toHaveStyleRule('background-color', 'rgb(117, 12, 12)');
+      });
+    });
+
+    describe('Style overrides using a custom element name', () => {
+      it('Should apply correct style rules', () => {
+        render(<BaseMenu menuBadgeVariant="decorative10" element="CUSTOM" />, {
+          wrapper: initWrapper('custom'),
+        });
+
+        const nodeCalledMenuButton = screen.getByTestId('menu-badge');
+
+        expect(nodeCalledMenuButton).toHaveStyleRule('background-color', 'rgb(117, 12, 12)');
+      });
+    });
+  });
+});
+
+const NOOP = (): null => null;
+
+jest.mock('../src/useMenuState', () => {
+  return {
+    useMenuState: (baseId: string): Record<string, any> => ({
+      animated: false,
+      animating: false,
+      baseId,
+      currentId: null,
+      down: NOOP,
+      first: NOOP,
+      groups: [],
+      hide: NOOP,
+      items: [
+        {disabled: false, groupId: undefined, id: 'primary-menu-customization-story-1', ref: {current: 'button'}},
+        {disabled: false, groupId: undefined, id: 'primary-menu-customization-story-2', ref: {}},
+        {disabled: false, groupId: undefined, id: 'primary-menu-customization-story-3', ref: {}},
+        {disabled: false, groupId: undefined, id: 'primary-menu-customization-story-4', ref: {}},
+        {disabled: false, groupId: undefined, id: 'primary-menu-customization-story-5', ref: {}},
+        {disabled: false, groupId: undefined, id: 'primary-menu-customization-story-6', ref: {}},
+      ],
+      last: NOOP,
+      loop: false,
+      modal: false,
+      move: NOOP,
+      next: NOOP,
+      orientation: 'vertical',
+      place: NOOP,
+      placement: 'bottom-start',
+      previous: NOOP,
+      registerGroup: NOOP,
+      registerItem: NOOP,
+      reset: NOOP,
+      rtl: false,
+      setAnimated: NOOP,
+      setBaseId: NOOP,
+      setCurrentId: NOOP,
+      setLoop: NOOP,
```

Comment:
0_0

Reply:
It's pure copy and paste from existing tests from Menu